src/main/java/meteordevelopment/meteorclient/gui/screens/settings/ColorSettingScreen.java

```diff
@@ -58,14 +58,12 @@ public boolean fromClipboard() {
         SettingColor parsed;
 
         if ((parsed = parseRGBA(clipboard)) != null) {
-            setting.get().rainbow(false);
             setting.set(parsed);
             setting.get().validate();
             return true;
         }
 
         if ((parsed = parseHex(clipboard)) != null) {
-            setting.get().rainbow(false);
```

Comment:
This was done on purpose, when you are pasting a color into the color picker you don't want to have rainbow enable from before, that doesn't make sense as the pasted color is never gonna be used anyways.

Reply:
the `Setting`’s value isn’t mutable, running `setting.set(parsed)` completely overwrites the previous value, so setting a value prior to overwriting it is useless

I think you’re confusing
`setting.set(parsed)` with `setting.get().set(parsed)` in which case that line would be required as it doesn’t take the rainbow parameter into account

Reply:
Oh yeah, sorry my bad